src/main/java/seedu/address/logic/parser/AddAdminCommandParser.java

```diff
@@ -31,29 +36,31 @@ public AddAdminCommand parse(String args) throws ParseException {
                 ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_USERNAME,
                         PREFIX_PASSWORD, PREFIX_PATH_TO_PIC , PREFIX_SALARY, PREFIX_EMPLOYMENT_DATE);
 
-        if (!arePrefixesPresent(argMultimap, PREFIX_USERNAME, PREFIX_PASSWORD, PREFIX_SALARY, PREFIX_EMPLOYMENT_DATE)
+        if (!arePrefixesPresent(argMultimap, PREFIX_USERNAME, PREFIX_PASSWORD, PREFIX_SALARY, PREFIX_EMPLOYMENT_DATE,
+                PREFIX_NAME, PREFIX_PATH_TO_PIC)
                 || !argMultimap.getPreamble().isEmpty()) {
             throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, AddAdminCommand.MESSAGE_USAGE));
         }
 
 
         //Admin Credential
-        String username = argMultimap.getValue(PREFIX_USERNAME).get();
+        Username username = ParserUtil.parseUsername(argMultimap.getValue(PREFIX_USERNAME).get());
         String password = argMultimap.getValue(PREFIX_PASSWORD).get();
 
         //Admin info
-        String name = argMultimap.getValue(PREFIX_NAME).get();
+        Name name = ParserUtil.parseName(argMultimap.getValue(PREFIX_NAME).get());
         Role role = Role.ADMIN;
-        String pathToProfilePic = argMultimap.getValue(PREFIX_PATH_TO_PIC).get();
-        int salary = Integer.parseInt(argMultimap.getValue(PREFIX_SALARY).get());
-        String employmentDate = argMultimap.getValue(PREFIX_PATH_TO_PIC).get();
+        PathToProfilePic pathToProfilePic =
+                ParserUtil.parsePathToProfilePic(argMultimap.getValue(PREFIX_PATH_TO_PIC).get());
+        Salary salary = ParserUtil.parseSalary(argMultimap.getValue(PREFIX_SALARY).get());
+        EmployDate employmentDate = ParserUtil.parseEmployDate(argMultimap.getValue(PREFIX_EMPLOYMENT_DATE).get());
         Admin admin = new Admin(username, name, role, pathToProfilePic, salary,
             employmentDate);
 
 
 
         //TODO key to be replaced
```

Comment:
Is this comment still required?

Reply:
Not anymore, since key will be the same as password. It has been removed in the latest commit.